app/models/atlas.rb

```diff
@@ -0,0 +1,32 @@
+class Atlas < ActiveRecord::Base
+  belongs_to :taxon
+  has_many :exploded_atlas_places
+  has_many :atlas_alterations
+  
+  def places
+    exploded_place_ids_to_include, exploded_place_ids_to_exclude = get_exploded_place_ids_to_include_and_exclude
+    Place.
+      where( "(admin_level = 0 OR places.id IN (?))", exploded_place_ids_to_include ).
+      where( "places.id NOT IN (?)", exploded_place_ids_to_exclude )
+  end
+  
+  def presence_places
+    exploded_place_ids_to_include, exploded_place_ids_to_exclude = get_exploded_place_ids_to_include_and_exclude
+    Place.joins( listed_taxa: :list ).
+      where( "lists.type = 'CheckList'").
+      where( "lists.id = places.check_list_id" ).
+      where( "listed_taxa.taxon_id = ?", taxon ).
+      where( "(admin_level = 0 OR places.id IN (?))", exploded_place_ids_to_include ).
+      where( "places.id NOT IN (?)", exploded_place_ids_to_exclude )
+  end
+  
+  def get_exploded_place_ids_to_include_and_exclude
+    exploded_place_ids_to_include = [0]
+    exploded_place_ids_to_exclude = [0]
```

Comment:
There should never be a place with an ID of 0, so your code will work. Another approach would be to just check whether those arrays are blank and alter the query accordingly, e.g.

```ruby
places = Place.
  joins( listed_taxa: :list ).
  where( "lists.type = 'CheckList'").
  where( "lists.id = places.check_list_id" )
  where( "listed_taxa.taxon_id = ?", taxon )
places = if exploded_place_ids_to_include.blank?
  places.where( admin_level: 0 )
else
  places.where( "(admin_level = 0 OR places.id IN (?))", exploded_place_ids_to_include ).
end
unless exploded_place_ids_to_exclude.blank?
  places = places.where( "places.id NOT IN (?)", exploded_place_ids_to_exclude )
end
places
```